pyuvdata/uvcal/uvcal.py

```diff
@@ -380,7 +380,8 @@ def __init__(self):
         )
 
         desc = (
-            "Required if flex_spw = True. Maps individual channels along the "
+            "Required if flex_spw = True and will always be required for non-wide-band "
+            "objects starting in version 3.0. Maps individual channels along the "
```

Comment:
Does the `flex_spw_id_array` have an obvious default? It seems that in many applications, it'll just be all zeros. Can we make it have a default?

Reply:
There's an obvious default if there is only one spectral window, in which case the obvious default is to fill it with the (single) value in the `spw_array`.

Reply:
I added this defaulting to all the `read` methods, so users who just read in files will not notice any change. Do you want me to add defaulting to the `check` method as well?

Reply:
I think this is another thing you'd default sensibly in a method that builds an object from the "minimal" data set. I suppose we could build in a default using `__getattr__` machinery, but that can get a bit complicated (as I've learned in pyradiosky). For now, it's just a warning, but it's heading towards being required.

Reply:
No, let's not do a `__getattr__`! I think this is fine -- we can build it into a `.new()` classmethod, or whatever we call it.